CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+             * Decision for which hand wins in a tie is certainly overtuned but should be complete, 
+             *  and operates as if only Ranking.HIGH_CARD is possible.
+             * 
+             * Algorithm is simply :=   Score = 5^hand[4] + 4^hand[3] + ... + 1^hand[0];
+             * 12 12 12 12 12 =>    261,453,379
+             * 13 1  1  1  1  =>    1,220,703,135
+             * Which is what we want.
+             * 
+             * AAAAK    => 6,373,561,789
+             * INT_MAX  => 2,147,483,647
+             * So we will use 64 bit integer.
+             */
+            long handVal = 0;
+            for (int j = 4; j >= 0; j++)
+            {
+                handVal += (long)Math.Pow((j + 1), hand[j].GetPokerValue());
+            }
+            return handVal;
+        }
+
+
+        /// <summary>
+        /// Function to determine if a poker hand is a royal flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a royal flush, false otherwise.</returns>
+        private static bool IsRoyalFlush(List<Card> cards)
+        {
+            return IsStraightFlush(cards) && cards[0].GetPokerValue() == 10;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a flush, false otherwise.</returns>
+        private static bool IsFlush(List<Card> cards)
+        {
+            Suit suit = cards[0].Suit;
+            foreach (Card card in cards)
+            {
+                if (card.Suit != suit)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a straight flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight flush, false otherwise.</returns>
+        private static bool IsStraightFlush(List<Card> cards)
+        {
+            return IsFlush(cards) && IsStraight(cards);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a four of a kind.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a four of a kind, false otherwise.</returns>
+        private static bool IsFourOfAKind(List<Card> cards)
+        {
+            return cards[0].SameVal(cards[3]);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a full house.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a full house, false otherwise.</returns>
+        private static bool IsFullHouse(List<Card> cards)
+        {
+            if (!cards[0].SameVal(cards[1]) || !cards[3].SameVal(cards[4]))
```

Comment:
Same here. Use `cards[0] != cards[1]` and so on.

This can also be reduced into a single return statement:

`return cards[0] == cards[1] && cards[3] == cards[4] && (cards[2] == cards[1] || cards[2] == cards[3]);`